src/pages/home/report/comment/TextCommentFragment.js

```diff
@@ -0,0 +1,114 @@
+import Str from 'expensify-common/lib/str';
+import PropTypes from 'prop-types';
+import React, {memo} from 'react';
+import Text from '@components/Text';
+import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
+import withWindowDimensions, {windowDimensionsPropTypes} from '@components/withWindowDimensions';
+import ZeroWidthView from '@components/ZeroWidthView';
+import compose from '@libs/compose';
+import convertToLTR from '@libs/convertToLTR';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as EmojiUtils from '@libs/EmojiUtils';
+import reportActionFragmentPropTypes from '@pages/home/report/reportActionFragmentPropTypes';
+import reportActionSourcePropType from '@pages/home/report/reportActionSourcePropType';
+import editedLabelStyles from '@styles/editedLabelStyles';
+import styles from '@styles/styles';
+import themeColors from '@styles/themes/default';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import RenderCommentHTML from './RenderCommentHTML';
+
+const propTypes = {
+    /** The reportAction's source */
+    source: reportActionSourcePropType.isRequired,
+
+    /** The message fragment needing to be displayed */
+    fragment: reportActionFragmentPropTypes.isRequired,
+
+    /** Should this message fragment be styled as deleted? */
+    styleAsDeleted: PropTypes.bool.isRequired,
+
+    /** Text of an IOU report action */
+    iouMessage: PropTypes.string,
+
+    /** Should the comment have the appearance of being grouped with the previous comment? */
+    displayAsGroup: PropTypes.bool.isRequired,
+
+    /** Additional styles to add after local styles. */
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]).isRequired,
+
+    ...windowDimensionsPropTypes,
+
+    /** localization props */
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    iouMessage: undefined,
+};
+
+function TextCommentFragment(props) {
+    const {fragment, styleAsDeleted} = props;
+    const {html, text} = fragment;
+
+    // If the only difference between fragment.text and fragment.html is <br /> tags
+    // we render it as text, not as html.
+    // This is done to render emojis with line breaks between them as text.
+    const differByLineBreaksOnly = Str.replaceAll(html, '<br />', '\n') === text;
+
+    // Only render HTML if we have html in the fragment
+    if (!differByLineBreaksOnly) {
+        const editedTag = fragment.isEdited ? `<edited ${styleAsDeleted ? 'deleted' : ''}></edited>` : '';
+        const htmlContent = styleAsDeleted ? `<del>${html}</del>` : html;
+
+        const htmlWithTag = editedTag ? `${htmlContent}${editedTag}` : htmlContent;
+
+        return (
+            <RenderCommentHTML
+                source={props.source}
+                html={htmlWithTag}
+            />
+        );
+    }
+
+    const containsOnlyEmojis = EmojiUtils.containsOnlyEmojis(text);
+
+    return (
```

Comment:
What is the difference between what is rendered here and what is rendered in `RenderCommentHTML`? It feels like all the stuff here should also be in its own component.

Reply:
> What is the difference between what is rendered here and what is rendered in RenderCommentHTML?

Uh, here we render a bunch of text, and there we render HTML? 😁

> It feels like all the stuff here should also be in its own component.

It's already extracted from [ReportActionItemFragment.js](https://github.com/Expensify/App/pull/25524/files#diff-7451957e0135ee20216b214d4e27202271aeacbd2e321b9b35b5b1e60d6677b6) (you can check the red diff).

`RenderCommentHTML` was extracted because it is used at least twice. A pragmatic thing.